app/models/user.rb

```diff
@@ -0,0 +1,45 @@
+require_relative '../data_mapper_setup'
+require 'bcrypt'
+
+class User
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :name, Text
+  property :username, Text,
+          :messages => {
+             :presence  => "Please enter a username address.",
+             :is_unique => "Sorry, that usename is not available."
+             }
+  validates_presence_of :username
+  validates_uniqueness_of :username
+   
+
+  property :password_digest, Text
+  validates_confirmation_of :password
+
+  property :email, String,
+           :format   => :email_address,
```

Comment:
Style/HashSyntax: Use the new Ruby 1.9 hash syntax. (https://github.com/bbatsov/ruby-style-guide#hash-literals)